basic_grunt.py

```diff
@@ -55,5 +56,20 @@ def take_damage(self, damage):
             print("Disregarding non positive damage")
         elif damage >= self.health_:
             self.health_ = 0
+            self.report_destroyed()
```

Comment:
Since in this statement, object will definitely be destroyed. Is it even worth it to have `self.health_ = 0` anymore? Alternatively, we could put this report_destroyed functionality only once in the code, at the bottom of the update. So at the end of this object's update function, it will have 
`if self.health <= 0:`
`self.report_destroyed()`